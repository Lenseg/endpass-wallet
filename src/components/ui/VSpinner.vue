<template>
  <div v-if="isLoading" class="spinner">
    <spinner :status="isLoading" :color="color" />
  </div>
</template>

<script>
import Spinner from 'vue-spinner-component/src/Spinner';

export default {
  name: 'v-spinner',
  data() {
    return {
      color: '#4b0472' // scss $primary color
    };
  },
  props: {
    isLoading: {
      type: Boolean,
      default: false,
    },
  },
  components: { Spinner }
};
</script>

<style lang="scss" scoped>
  .spinner {
    display: flex;
    justify-content: center;
    align-items: center;
  }
</style>
