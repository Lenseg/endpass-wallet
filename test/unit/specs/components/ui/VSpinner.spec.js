import { shallow } from '@vue/test-utils';

import VSpinner from '@/components/ui/VSpinner';

describe('VSpinner', () => {
  describe('render', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(VSpinner);
    });

    it('should be a Vue component', () => {
      expect(wrapper.name()).toBe('v-spinner');
      expect(wrapper.isVueInstance()).toBeTruthy();
    });
  });

  describe('props', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(VSpinner, {
        stubs: {
          spinner: '<spinner/>',
        },
      });
    });

    describe('isLoading', () => {
      it('should render component', () => {
        wrapper.setProps({ isLoading: true });

        expect(wrapper.element).toMatchSnapshot();
      });

      it('should not render component', () => {
        wrapper.setProps({ isLoading: false });

        expect(wrapper.element).toMatchSnapshot();
      });
    });
  });
});
