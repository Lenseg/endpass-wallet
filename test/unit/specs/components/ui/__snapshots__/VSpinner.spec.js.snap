// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VSpinner props isLoading should not render component 1`] = `<!---->`;

exports[`VSpinner props isLoading should render component 1`] = `
<div
  class="spinner"
>
  <spinner
    color="#4b0472"
    status="true"
  />
</div>
`;
